AssetManagementTestProject/PageObj/ManageUserPage.cs

```diff
@@ -1,71 +1,58 @@
     public readonly string BtnStaffCode = "//span[contains(text(), 'Staff Code')]";
     public readonly string BtnFullName = "//span[contains(text(), 'Full Name')]";
     public readonly string BtnUsername = "//span[contains(text(), 'Username')]";
     public readonly string BtnJoinedDate = "//span[contains(text(), 'Joined Date')]";
-    public readonly string BtnSortType = "(//span[text()= 'Type' and @class='ant-table-column-title'])";
-    public readonly string BtnSortAdminType = "(//div[(text()= 'Admin') and (@class='ant-select-item-option-content')])";
-    public readonly string BtnSortStaffType = "(//div[(text()= 'Staff') and (@class='ant-select-item-option-content')])";
-    public readonly string BtnType = "(//span[contains(text(), 'Type')])[2]";
-    #endregion
-    public readonly string HeaderDetailedUser = "//h1[text()='Detail User Information']";
-    public readonly string CellsDetailedInfo = "//td[contains(@class, 'font-bold')]/following-sibling::*";
-    #region SEARCH
-    public readonly string TfSearch = "//input[contains(@class,'ant-input css-1wismvm')]";
-    public readonly string BtnSearch = "//button[contains(@class,'ant-btn css-1wismvm ant-btn-default ant-btn-icon-only ant-input-search-button')]";
-    public readonly string TableData = "//tbody[contains(@class,'ant-table-tbody')]";
-    #endregion
-    #region GRID
-    public UserActualData? UserActualData;
-    public ViewUserInList? UserInfo;
-    public ViewDetailedUser? DetailedUserInfo;
-    #endregion
+    public readonly string BtnSortType = "//div[@class='ant-select w-3/12 css-1wismvm ant-select-single ant-select-allow-clear ant-select-show-arrow']";
+    public readonly string BtnSortAdminType = "//div[(text()= 'Admin') and (@class='ant-select-item-option-content')]";
+    public readonly string BtnSortStaffType = "//div[(text()= 'Staff') and (@class='ant-select-item-option-content')]";
+    public readonly string BtnType = "//span[text()= 'Type' and @class='ant-table-column-title']";
     public ManageUserPage() : base()
     {
     }
-
-    public void GoToUserList()
-    {
-        Click(BtnManageUser);
-    }
-
     public void InputSearch(string input)
     {
         SendKeys(TfSearch, input);
-        Click(BtnSearch);
+        FindElementByXpath(BtnSearch).Click();
+        /// Search is not fast enough
         Thread.Sleep(5000);
     }
-
+    public string ReturnStaffCodeTopListUser()
+    {
+        return FindElementByXpath(BtnViewTopUserDetailedInfo).Text;
+    }
     public void SelectAdminType()
     {
         FindElementByXpath(BtnSortType).Click();
-        FindElementByXpath(BtnSortAdminType).Click();
+        Click(BtnSortAdminType);
     }
     public void SelectStaffType()
     {
         FindElementByXpath(BtnSortType).Click();
-        FindElementByXpath(BtnSortAdminType).Click();
+        Click(BtnSortStaffType);
+    }
+    public void SortUser(string sortType)
+    {
+        Click(sortType);
+        Thread.Sleep(5000);
```

Comment:
Never use Thread sleep if you can => Try to wait for the loading icon disappear.
the hard sleep time should have a function to cover. For examples, waitForSmtAppear

Reply:
Sorry for deleting the comment that explains why I used thread sleep here. I have tried to use explicit wait for some locators such as first row of the table, edit button,... but somehow it always stop the search and edit the user BEFORE search (which is the Admin Hanoi user). 
So far I have resorted to using thread sleep, but when I find an alternative I will update ASAP. I will make a seperate method for thread sleep too